docstore/driver/codec.go

```diff
@@ -510,7 +512,12 @@ func decodeList(v reflect.Value, d Decoder) error {
 	// supports that, then do the decoding.
 	if v.Type().Elem().Kind() == reflect.Uint8 {
 		if b, ok := d.AsBytes(); ok {
-			v.SetBytes(b)
+			if v.Kind() == reflect.Slice {
+				v.SetBytes(b)
```

Comment:
This would previously panic with `reflect: call of reflect.Value.SetBytes on array Value`.

Reply:
The check for slice is good, but if false I think you want to fall through to the general list-handling code on line 536/551. It should handle copying into arrays of various size, not just the same size. Currently you will not assign if the destination array is a different size, which is a bug.

Reply:
Hmm, that does not work. If I let it fall through, the call to `d.ListLen()` below fails.

Reply:
OK, keep your code and return an error if the byte array is a different size, maybe with a TODO that it should work like the general array-copying code below.

Reply:
I modified it to call `prepareLen` it if it's an array, I think this does the right thing but PTAL.

Reply:
LGTM.